scalameta/quasiquotes/shared/src/main/scala-3/scala/meta/internal/quasiquotes/ReificationMacros.scala

```diff
@@ -0,0 +1,688 @@
+                    List(TypeTree.of[T])
+                  ),
+                  Nil,
+                  List(liftTree(otherTree))
+                ),
+                TypeTree.of[Some[T]]
+              )
+          case None => '{ scala.None }.asTerm match
+              case Inlined(_, _, none) => none
+        }
+      def liftTrees[T: Type](trees: Seq[MetaTree]): Tree = {
+        @tailrec
+        def loop(trees: Seq[MetaTree], acc: Option[Tree], prefix: List[MetaTree]): Tree =
+          trees match {
+            case (quasi: Quasi) +: rest if quasi.rank == 1 =>
+              if (acc.isEmpty)
+                if (prefix.isEmpty) loop(rest, Some(liftQuasi0(quasi)), Nil)
+                else loop(
+                  rest,
+                  prefix.foldRight(acc) { (curr, acc) =>
+                    // Note copied from the Scala 2 counterpart file.
+                    // NOTE: We cannot do just q"${liftTree(curr)} +: ${liftQuasi(quasi)}"
+                    // because that creates a synthetic temp variable that doesn't make any sense in a pattern.
+                    // Neither can we do q"${liftQuasi(quasi)}.+:(${liftTree(curr)})",
+                    // because that still wouldn't work in pattern mode.
+                    // Finally, we can't do something like q"+:(${liftQuasi(quasi)}, (${liftTree(curr)}))",
+                    // because would violate evaluation order guarantees that we must keep.
+                    val currElement = liftTree(curr)
+                    val alreadyLiftedList = acc.getOrElse(liftQuasi(quasi))
+                    val tree =
+                      if (mode.isTerm) Apply(
+                        TypeApply(
+                          Select.unique(alreadyLiftedList.asInstanceOf[Term], "+:"),
+                          List(TypeTree.of[T])
+                        ),
+                        List(currElement.asInstanceOf[Term])
+                      )
+                      else Unapply(
+                        TypeApply(
+                          Select.unique('{ +: }.asTerm, "unapply"),
+                          List(TypeTree.of[T], TypeTree.of[List], TypeTree.of[List[T]])
+                        ),
+                        Nil,
+                        List(currElement, alreadyLiftedList)
+                      )
+                    Some(tree)
+                  },
+                  Nil
+                )
+              else {
+                Predef.require(prefix.isEmpty && debug(trees, acc, prefix))
+                if mode.isTerm then
+                  val tree = Apply(
+                    TypeApply(Select.unique(acc.get.asInstanceOf[Term], "++"), List(TypeTree.of[T])),
+                    List(liftQuasi(quasi).asInstanceOf[Term])
+                  )
+                  loop(rest, Some(tree), Nil)
+                else report.errorAndAbort(Messages.QuasiquoteAdjacentEllipsesInPattern(quasi.rank))
+              }
```

Comment:
I would strongly prefer to avoid `Option.get` (like you do on line 389 here, and in some other parts of the code as well). why not:

```
              acc match {
                case None if prefix.isEmpty => loop(rest, Some(liftQuasi0(quasi)), Nil)
                case None =>
                  val acc = prefix.foldRight(Option.empty[Tree]) { (curr, acc) =>
                    // Note copied from the Scala 2 counterpart file.
                    // NOTE: We cannot do just q"${liftTree(curr)} +: ${liftQuasi(quasi)}"
                    // because that creates a synthetic temp variable that doesn't make any sense in a pattern.
                    // Neither can we do q"${liftQuasi(quasi)}.+:(${liftTree(curr)})",
                    // because that still wouldn't work in pattern mode.
                    // Finally, we can't do something like q"+:(${liftQuasi(quasi)}, (${liftTree(curr)}))",
                    // because would violate evaluation order guarantees that we must keep.
                    val currElement = liftTree(curr)
                    val alreadyLiftedList = acc.getOrElse(liftQuasi(quasi))
                    Some(
                      if (mode.isTerm) Apply(
                        TypeApply(
                          Select.unique(alreadyLiftedList.asInstanceOf[Term], "+:"),
                          List(TypeTree.of[T])
                        ),
                        List(currElement.asInstanceOf[Term])
                      )
                      else Unapply(
                        TypeApply(
                          Select.unique('{ +: }.asTerm, "unapply"),
                          List(TypeTree.of[T], TypeTree.of[List], TypeTree.of[List[T]])
                        ),
                        Nil,
                        List(currElement, alreadyLiftedList)
                      )
                    )
                  }
                  loop(rest, acc, Nil)
                case Some(x: Term) if mode.isTerm =>
                  Predef.require(prefix.isEmpty && debug(trees, acc, prefix))
                  val tree = Apply(
                    TypeApply(Select.unique(x, "++"), List(TypeTree.of[T])),
                    List(liftQuasi(quasi).asInstanceOf[Term])
                  )
                  loop(rest, Some(tree), Nil)
                case _ =>
                  report.errorAndAbort(Messages.QuasiquoteAdjacentEllipsesInPattern(quasi.rank))
              }
```